unittests/snapshot_tests.cpp

```diff
@@ -61,6 +61,7 @@ class snapshotted_tester : public base_tester {
       controller::config copied_config = (copy_files_from_config == copy_config_files)
                                          ? copy_config_and_files(config, ordinal) : copy_config(config, ordinal);
 
+      BOOST_CHECK_GT(snapshot->total_row_count(), 0);
```

Comment:
This is good enough to make sure all three impls of `total_row_count()` are sane (and indeed I found my rapidjson impl was incorrect initially). I don't think we need to actually make sure the exact count is accurate and precise.